windows builds: devenv freetype /update appears not to have an effect

Despite attempting to upgrade the freetype `sln` file with `/upgrade`, when building using the 2017 `msbuild` as part of `setup.py` I get the following failure:
```
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
    Build started 2020-01-16 14:46:40.
    Project "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.sln" on node 1 (Clean;Build target(s)).
    ValidateSolutionConfiguration:
      Building solution configuration "Release|x64".
    Project "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.sln" (1) is building "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.vcxproj" (2) on node 1 (Clean target(s)).
    CoreClean:
      Creating directory ".\..\..\..\objs\vc2010\x64\Release\".
    Done Building Project "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.vcxproj" (Clean target(s)).
    Project "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.sln" (1) is building "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.vcxproj" (2:2) on node 1 (default targets).
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Platform.targets(55,5): error MSB8020: The build tools for Visual Studio 2010 (Platform Toolset = 'v100') cannot be found. To build using the v100 build tools, please install Visual Studio 2010 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.vcxproj]
    Done Building Project "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.vcxproj" (default targets) -- FAILED.
    Done Building Project "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.sln" (Clean;Build target(s)) -- FAILED.

    Build FAILED.

    "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.sln" (Clean;Build target) (1) ->
    "C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.vcxproj" (default target) (2:2) ->
    (PlatformPrepareForBuild target) ->
      C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Platform.targets(55,5): error MSB8020: The build tools for Visual Studio 2010 (Platform Toolset = 'v100') cannot be found. To build using the v100 build tools, please install Visual Studio 2010 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\Users\wiese\Repos\numeric-python\matplotlib\build\freetype-2.6.1\builds\windows\vc2010\freetype.vcxproj]
```

--- a/setupext.py
+++ b/setupext.py
@@ -561,9 +561,22 @@ class FreeType(SetupPackage):
  </PropertyGroup>
 </Project>
 """)
+            # It is not a trivial task to determine PlatformToolset to plug it
+            # into msbuild command, and Directory.Build.props will not override
+            # the value in the project file.
+            # The DefaultPlatformToolset is from Microsoft.Cpp.Default.props
+            with open(base_path / vc / "freetype.vcxproj", 'r+b') as f:
+                toolset_repl = b'PlatformToolset>$(DefaultPlatformToolset)<'
+                vcxproj = f.read().replace(b'PlatformToolset>v100<',
+                                           toolset_repl)
+                assert toolset_repl in vcxproj, (
+                   'Upgrading Freetype might break this')
+                f.seek(0)
+                f.truncate()
+                f.write(vcxproj)
+
             cc = ccompiler.new_compiler()
-            cc.initialize()  # Get devenv & msbuild in the %PATH% of cc.spawn.
-            cc.spawn(["devenv", str(sln_path), "/upgrade"])
+            cc.initialize()  # Get msbuild in the %PATH% of cc.spawn.
             cc.spawn(["msbuild", str(sln_path),
                       "/t:Clean;Build",
                       f"/p:Configuration=Release;Platform={msbuild_platform}"])

